hcloud/resource_hcloud_reversedns.go

```diff
@@ -0,0 +1,226 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+func resourceReverseDNS() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceReverseDNSRead,
+		Create: resourceReverseDNSCreate,
+		Delete: resourceReverseDNSDelete,
+
+		Schema: map[string]*schema.Schema{
+			"server_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"floating_ip_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"dns_ptr": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceReverseDNSRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ipAddress := string(d.Get("ip_address").(string))
+	server, floatingIP, err := lookupRDNSID(d.Id(), client)
+
+	switch {
+	case err != nil:
+		log.Printf("[WARN] Invalid id (%s), removing from state: %s", d.Id(), err)
+		d.SetId("")
+		return nil
+	}
+	ip := net.ParseIP(d.Get("ip_address").(string))
+	if ip == nil {
+		log.Printf("[ERR] The ip you provide (%s) is not valid", ipAddress)
+		return nil
+	}
+	switch {
+	case floatingIP != nil:
+		if floatingIP == nil {
+			log.Printf("[WARN] Floating IP (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		ip = net.ParseIP(ipAddress)
+		switch floatingIP.Type {
+		case "ipv4":
+			if ip.To4() == nil {
+				log.Printf("[WARN] Floating IP (%s) is an ipv4 floating ip but you dont write a valid one, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+		case "ipv6":
+			if ip.To16() == nil {
+				log.Printf("[WARN] Floating IP (%s) is an ipv6 but you write an invalid ip, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+		}
+		dnsPtr := floatingIP.DNSPtrForIP(ip)
+		if dnsPtr != "" {
+			d.Set("dnsPtr", dnsPtr)
```

Comment:
You also have to `.Set()` all other `non ID` properties otherwise `terraform import` can not work.